src/app/Dashboard/components/DashboardComponent.js

```diff
@@ -42,49 +46,80 @@ export class DashboardComponent extends Component {
     } else {
       dashboardHtml = (
-        <div className='user-dashboard'>
-          <h2 className='user-dashboard__name col-sm-12'>{user.name}</h2>
-          <div className='col-sm-12'>
-            <h3 className='user-dashboard__level'>D2</h3>
-          </div>
-          <div className='profile-overview col-sm-12'>
-            <div className='profile-overview__image' />
-            <MyStatsComponent points={pointsEarned} activities={activitiesLogged} />
-            <SocietyStatsComponent society={society} usedPoints={1508} remainingPoints={326} />
-          </div>
-          <div className='user-dashboard__actions col-sm-12'>
-            <h3 className='user-dashboard__title'>My Activities</h3>
-            <div>
-              <ButtonComponent className='button__add'>
-                <span className='fa fa-plus' />
-                <span>Log Points</span>
-              </ButtonComponent>
-              <ButtonComponent className='button__filter'>
-                <span>Filter</span>
-                <span className='fa fa-filter' />
-              </ButtonComponent>
+        <div className='wrapper'>
+          <HeroComponent />
+          <div className='main-content'>
+            <SidebarComponent className='sidebar' />
+            <div className='sub-content'>
+              <NavbarComponent />
+              <div className='user-dashboard'>
+                <h2 className='user-dashboard__name col-sm-12'>{user.name}</h2>
+                <div className='col-sm-12'>
+                  <h3 className='user-dashboard__level'>D2</h3>
+                </div>
+                <div className='profile-overview col-sm-12'>
+                  <div className='profile-overview__image' />
+                  <MyStatsComponent points={pointsEarned} activities={activitiesLogged} />
+                  <SocietyStatsComponent usedPoints={1508} remainingPoints={326} />
+                </div>
+                <div className='user-dashboard__actions col-sm-12'>
+                  <h3 className='user-dashboard__title'>My Activities</h3>
+                  <div>
+                    {/* <ButtonComponent className='button__add'>
+                      <span className='fa fa-plus' />
+                      <span>Log Points</span>
+                    </ButtonComponent> */}
+                    <button type='button' onClick={this.openModalLoginPointsHandler}>Log-in points</button>
+                    <ButtonComponent className='button__filter' onClick={this.openModalLoginPointsHandler}>
+                      <span>Filter</span>
+                      <span className='fa fa-filter' />
+                    </ButtonComponent>
+                    {
+                      logPoints
+                        ? (
+                          <LoginModal
+                            className='modal'
+                            show={logPoints}
+                          />
+                        )
+                        : null
+                    }
```

Comment:
I think we have talked about this enough times. Evaluate the JSX before the render. This becomes unmanageable

Reply:
@kimani-ndegwa Thanks for catching this. I have removed the evaluation it was not even necessary.